src/lib/quoter.js

```diff
@@ -52,9 +52,14 @@ class Quoter {
     const hop = this._findBestHopForAmount(request)
     if (!hop) return null
 
-    // If we know a local route to the destinationAccount, use the local route.
+    // note the confusion between `next`, `destination`, and `final`;
+    // `finalAmount` will be renamed to `destinationAmount` later in
+    // the hopToQuote function.
+    if (hop.sourceAmount && hop.finalAmount) {
```

Comment:
If the `hop.sourceAmount`/`hop.finalAmount` is `0`, then yes, it should still return `hopToQuote(hop, destinationHoldDuration)`.